src/preact.d.ts

```diff
@@ -4,8 +4,8 @@ export as namespace preact;
 declare namespace preact {
 	type Key = string | number;
 	type Ref<T> = (instance: T) => void;
-	type ComponentChild = VNode<any> | string | number | null;
-	type ComponentChildren = ComponentChild[] | ComponentChild | object | string | number | null;
```

Comment:
The tests are failing because the `object` type is needed to get more complex children to typecheck properly with jsx. See #1116 .